rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/policy/RestartDownNodesPolicy.java

```diff
@@ -47,73 +47,102 @@
 
     private static Logger logger = Logger.getLogger(RestartDownNodesPolicy.class);
 
-    private Timer timer = new Timer("RestartDownNodesPolicy node status check");
+    private static final String TIMER_NAME = "RestartDownNodesPolicy node status check";
 
-    @Configurable(description = "ms (30 mins by default)")
+    @Configurable(description = "ms (30 mins by default)", dynamic = true)
     private int checkNodeStateEach = 30 * 60 * 1000; // 30 mins by default;
 
+    private Timer timer;
+
     /**
      * Configure a policy with given parameters.
      * @param policyParameters parameters defined by user
      */
     @Override
     public BooleanWrapper configure(Object... policyParameters) {
+
         BooleanWrapper parentConfigured = super.configure(policyParameters);
+
         if (policyParameters != null && policyParameters.length > 2) {
-            checkNodeStateEach = Integer.parseInt(policyParameters[2].toString());
+            this.checkNodeStateEach = Integer.parseInt(policyParameters[2].toString());
         }
+
+        this.timer = new Timer(TIMER_NAME);
+
         return parentConfigured;
     }
 
+    @Override
+    public void reconfigure(Object... policyParameters) {
+
+        this.timer.cancel();
+
+        this.configure(policyParameters);
```

Comment:
question, are we sure that this configure, which calls super.configure() will not trigger any side effects with non-dynamic fields ?
Isn't it better to only modify parameters that the policy knows as dynamic ?

Reply:
second question: what about database persistence, what will happen in case of:
reconfigure -> scheduler restart -> reload node source policy from db ?

Reply:
For the second question, in case of restart the last validated parameters of the infrastructure and of the policy will always be loaded (this is not linked with any resource manager property). 

Note that this is unrelated to nodes recovery, because we are speaking here about the configuration parameters of a node source, and not about the variables it uses to maintain its state at runtime.

Reply:
For the first question I changed both the default implementation and the implementation in RestartDownNodesPolicy that only lookup dynamic fields and set them only by reflection.